services/rollout-service/pkg/service/broadcast_test.go

```diff
@@ -437,3 +483,140 @@ func TestBroadcastDoesntGetStuck(t *testing.T) {
 
 	}
 }
+
+func TestGetStatus(t *testing.T) {
+	t.Parallel()
+
+	tcs := []struct {
+		Name            string
+		ArgoEvents      []ArgoEvent
+		KuberpultEvents []versions.KuberpultEvent
+		Request         *api.GetStatusRequest
+
+		ExpectedResponse *api.GetStatusResponse
+	}{
+		{
+			Name:    "simple case",
+			Request: &api.GetStatusRequest{},
+			ExpectedResponse: &api.GetStatusResponse{
+				Status: api.RolloutStatus_RolloutStatusSuccesful,
+			},
+		},
+		{
+			Name: "filters for environmentGroup",
+			ArgoEvents: []ArgoEvent{
+				{
+					Application:      "foo",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 2},
+					SyncStatusCode:   v1alpha1.SyncStatusCodeSynced,
+					HealthStatusCode: health.HealthStatusHealthy,
+				},
+				{
+					Application:      "foo",
+					Environment:      "prd",
+					Version:          &versions.VersionInfo{Version: 1},
+					SyncStatusCode:   v1alpha1.SyncStatusCodeSynced,
+					HealthStatusCode: health.HealthStatusHealthy,
+				},
+			},
+			KuberpultEvents: []versions.KuberpultEvent{
+				{
+					Application:      "foo",
+					Environment:      "dev",
+					Version:          &versions.VersionInfo{Version: 3},
+					EnvironmentGroup: "dev-group",
+				},
+				{
+					Application:      "foo",
+					Environment:      "prd",
+					Version:          &versions.VersionInfo{Version: 3},
+					EnvironmentGroup: "prd-group",
+				},
+			},
+			Request: &api.GetStatusRequest{
+				EnvironmentGroup: "dev-group",
+			},
+			ExpectedResponse: &api.GetStatusResponse{
+				Status: api.RolloutStatus_RolloutStatusPending,
+				Applications: []*api.GetStatusResponse_ApplicationStatus{
+					{
+						Environment:   "dev",
+						Application:   "foo",
+						RolloutStatus: api.RolloutStatus_RolloutStatusPending,
+					},
+				},
+			},
+		},
+		{
+			Name: "exludes succesful applications",
```

Comment:
I have my doubts here, I think I am not understanding everything here. Because the test data is the same and the response does not differ from the previous test case.

Reply:
Ah, yes, the result is the same but the test data is slightly different. In the second case, both applications are in the same env group, so the successful one is exlcuded.